qcodes/tests/test_parameter.py

```diff
@@ -1762,3 +1763,42 @@ def test_unknown_args_to_baseparameter_warns():
         a = _BaseParameter(name='Foo',
                            instrument=None,
                            snapshotable=False)
+
+
+@pytest.mark.parametrize("get_cmd, set_cmd", [(False, False), (False, None), (None, None), (None, False),
+                                              (lambda: 1, lambda x: x)])
+def test_gettable_settable_attributes_with_get_set_cmd(get_cmd, set_cmd):
+    a = Parameter(name='foo',
+                  get_cmd=get_cmd,
+                  set_cmd=set_cmd)
+    expected_gettable = get_cmd is not False
+    expected_settable = set_cmd is not False
+
+    assert a.gettable is expected_gettable
+    assert a.settable is expected_settable
+
+
+def test_gettable_settable_attributes_with_get_set_raw():
+    """Test that parameters that have get_raw,set_raw are
+    listed as gettable/settable and reverse."""
+
+    class GetSetParam(_BaseParameter):
```

Comment:
for completeness, shall we also add a subclass of a Parameter with get_raw/set_raw? it's great that Parameter already respects that _BaseParameter CAN create get/set methods (and we might have a test for that as well perhaps), but perhaps it's better to be a bit extra safe here?